File: tasks-service/TasksAG/src/main/java/com/example/TasksAG/domain/Task.java
package com.example.TasksAG.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String subject;
    private LocalDate duedate;
    private String status;
    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Client client;
    private boolean isDeleted;


    public String getClientEmail() {
        return client.getEmail();
    }

    public void setClientEmail(String email) {
        client.getEmail();
    }

    public Long getClientId() {
        return client.getId();
    }

    public void setClientId(Long clientId) {
        client.getId();
    }


}

File: tasks-ng/src/app/task.ts
export interface Task {
  id: number;
  subject: string;
  status: string;
  duedate: string;

}
export interface AllTask{
  id?:number;
  subject?: string;
  status?: string;
  duedate?: string;
  clientId?:number;
  clientEmail?:string;
}

export interface SearchForm {
  clientId:number | null;
  subject:string | null;
  duedate:string | null;
}
